metric-anomaly-data-model/src/main/proto/org/hypertrace/alert/engine/eventcondition/config/service/v1/metric_anomaly_event_condition.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.alert.engine.eventcondition.config.service.v1;
+
+message MetricAnomalyEventCondition {
+  MetricSelection metric_selection = 1;
+  repeated ViolationCondition violation_condition = 2;
+}
+
+message MetricSelection {
+  string key = 1;
+  string scope = 2;
+  MetricAggregationFunction metric_aggregation_function = 3;
+  string metric_aggregation_interval = 4; //iso duration string PT15S
+  repeated Filter filters = 5;
+}
+
+enum MetricAggregationFunction {
+  METRIC_AGGREGATION_FUNCTION_TYPE_UNSPECIFIED = 0;
+  METRIC_AGGREGATION_FUNCTION_TYPE_SUM = 1;
+  METRIC_AGGREGATION_FUNCTION_TYPE_AVG = 2;
+}
+
+message Filter {
+  oneof filter {
+    LeafFilter leaf_filter = 1;
+    CompositeFilter composite_filter = 2;
+  }
+}
+
+message LeafFilter {
+  LhsExpression lhs_expression = 1;
+  Operator operator = 2;
+  RhsExpression rhs_expression = 3;
+}
+
+enum Operator {
+  OPERATOR_UNSPECIFIED = 0;
+  OPERATOR_EQ = 1;
+  OPERATOR_AND = 2;
+  OPERATOR_OR = 3;
+}
+
+message CompositeFilter {
+  Operator operator = 1;
+  Filter lhs_filter = 2;
```

Comment:
a composite filter should have repeated children, I can AND/OR together any number of filters

Reply:
So, currently its modelled as a binary tree (operator as root and filter as left & right child). 
You are suggesting an n-ary tree representation ig, that will be closer to the existing filter definition in gateway and query (thus easier to convert)

Reply:
But, what will we gain? we are already able to express any number of AND/OR with this, right?

Reply:
yep - while binary can of course represent the same, filters are more naturally represented with a regular (that is, n-ary) tree.

Reply:
> But, what will we gain?

I can equally ask that about the binary representation :) Other than consistency with our other filter structures, a binary tree is just much more verbose to represent the same thing - why place that restriction?

Reply:
I was finding it more ease while reading it, but as it helps in impl, can do that as well. I have addressed this too.